mmdet/datasets/coco_occluded.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import mmcv
+import numpy as np
+from mmcv.fileio import load
+from mmcv.utils import print_log
+from pycocotools import mask as coco_mask
+from terminaltables import AsciiTable
+
+from .coco import CocoDataset
+
+
+class OccludedSeparatedCocoDataset(CocoDataset):
+    """COCO dataset with evaluation on separated and occluded masks which
+    presented in paper `A Tri-Layer Plugin to Improve Occluded Detection.
+
+    <https://arxiv.org/abs/2210.10046>`_.
+
+    Separated COCO and Occluded COCO are automatically generated subsets of
+    COCO val dataset, collecting separated objects and partially occluded
+    objects for a large variety of categories. In this way, we define
+    occlusion into two major categories: separated and partially occluded.
+    - Separation: target object segmentation mask is separated into distinct
+      regions by the occluder.
+    - Partial Occlusion: target object is partially occluded but the
+      segmentation mask is connected.
+    These two new scalable real-image datasets are to benchmark a model's
+    capability to detect occluded objects of 80 common categories.
+
+    Args:
+        occluded_ann (str): Path to the occluded coco annotation file.
+        separated_ann (str): Path to the separated coco annotation file.
+    """  # noqa
+
+    def __init__(
+            self,
+            *args,
+            occluded_ann='https://www.robots.ox.ac.uk/~vgg/research/tpod/datasets/occluded_coco.pkl',  # noqa
+            separated_ann='https://www.robots.ox.ac.uk/~vgg/research/tpod/datasets/separated_coco.pkl',  # noqa
+            **kwargs):
+        super().__init__(*args, **kwargs)
+        self.occluded_ann = load(occluded_ann)
```

Comment:
maybe this setting cannot handle the setting as local path, such as `data_root + 'annotations/xxx.pkl'`?